stages/eib_image

```diff
@@ -425,8 +425,43 @@ EOF
         umount "${DEPLOY}"/{boot,dev,img,proc,sys}
         rm -f "${DEPLOY}"/img
 
-        # Standard endless system - grub, shim, all that kind of thing
-        cp -r "${DEPLOY}"/usr/lib/efi_binaries/EFI "${ESP}"
+        # Install EFI binaries.
+        #
+        # In EOS6, the the binaries in `/usr/lib/efi_binaries` got there from
+        # `/boot/efi` in the initial Debian compose via eos-ostree-builder:
+        # <https://github.com/endlessm/eos-ostree-builder/blob/master/stages/eob_ostree#L76>
+        #
+        # The eos-build-meta GRUB elements install the GRUB x86_64-efi binary
+        # to same path ready to go in the ESP.
+        #
+        # For EOS7, we copy the Shim binary signed with Microsoft's certificate
+        # from Git. It's not gonna change in the short term as we would need
+        # to request a new signature from Microsoft.
+        #
+        # I also copied the mokmanager binary signed with Endless' certificate
+        # from an EOS6 system. This could instead be built in eos-build-meta
+        # and signed via the Endless secure-boot signing API.
+
+        # Install the "endless" UEFI bootloader entry.
+        mkdir -p "${ESP}/EFI/endless"
+        cp "${DEPLOY}"/usr/lib/efi_binaries/EFI/endless/grubx64.efi "${ESP}/EFI/endless/grubx64.efi"
+        cp "${EIB_DATADIR}"/shim-amd64/bootx64.csv "${ESP}/EFI/endless/BOOTX64.CSV"
+        cp "${EIB_DATADIR}"/shim-amd64/mmx64.efi "${ESP}/EFI/endless/mmx64.efi"
+        cp "${EIB_DATADIR}"/shim-amd64/shimx64-signed.efi "${ESP}/EFI/endless/shimx64.efi"
```

Comment:
This seems OK for the short term but ultimately wrong. If users are to be able to update shim on existing systems, then shim needs to be in the ostree.

Reply:
That will also require some mechanism for `ostree admin deploy` to get the new shim binary from wherever it is in the ostree and overwrite the ESP. As far as I know, nothing currently does that. (I may be wrong!)